src/deno_dir.rs

```diff
@@ -0,0 +1,121 @@
+use fs;
+use sha1;
+use std;
+use std::path::Path;
+use std::path::PathBuf;
+
+// Example: /Users/rld/.deno/
+static mut ROOT: Option<PathBuf> = None;
+static mut DEPS: Option<PathBuf> = None;
+static mut GEN: Option<PathBuf> = None;
+pub enum Dirname {
+  Root,
+  // In the Go code this was called SrcDir. Renaming to Deps.
+  // Example: /Users/rld/.deno/deps/
+  Deps,
+  // In the Go code this was called CacheDir. Renaming to GEN.
+  // Example: /Users/rld/.deno/gen/
+  Gen,
+}
+pub use self::Dirname::*;
+
+pub fn path(dirname: Dirname) -> &'static Path {
+  setup().expect("deno_dir setup failed.");
+  let pb_option = unsafe {
+    match dirname {
+      Root => &ROOT,
+      Deps => &DEPS,
+      Gen => &GEN,
+    }
+  };
+  match pb_option {
+    Some(ref x) => x.as_path(),
+    None => panic!(),
+  }
+}
+
+// https://github.com/ry/deno/blob/golang/deno_dir.go#L99-L111
+pub fn setup() -> std::io::Result<()> {
+  // Only setup once.
+  if unsafe { ROOT != None } {
+    return Ok(());
+  }
+  let home_dir = std::env::home_dir().expect("Could not get home directory.");
```

Comment:
Would rather convert `home_dir` Path/PathBuf here and get rid of the unsafe block below.

Reply:
It is a PathBuf here already https://doc.rust-lang.org/std/env/fn.home_dir.html

The unsafe is for the static global assignment.

Reply:
Ah, I didn't know this was considered unsafe